pkg/handler/scale_handler.go

```diff
@@ -514,33 +516,88 @@ func (h *ScaleHandler) getJobScalers(scaledObject *keda_v1alpha1.ScaledObject) (
 	return scalers, nil
 }
 
-func (h *ScaleHandler) getScaler(scaledObject *keda_v1alpha1.ScaledObject, trigger keda_v1alpha1.ScaleTriggers, resolvedEnv map[string]string) (scalers.Scaler, error) {
-	switch trigger.Type {
+func (h *ScaleHandler) resolveAuthSecret(name, namespace, key string) string {
+	if name == "" || namespace == "" || key == "" {
+		log.Errorf("name, namespace and key are required. Values: %s, %s, %s", name, namespace, key)
+		return ""
+	}
+
+	secret, err := h.kubeClient.CoreV1().Secrets(namespace).Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		log.Errorf("Error trying to get secret %s from namespace %s. Error: %s", name, namespace, err.Error())
+		return ""
+	}
+	result := secret.Data[key]
+
+	if result == nil {
+		return ""
+	}
+
+	return string(result)
+}
+
+func (h *ScaleHandler) parseAuthRef(triggerAuthRef keda_v1alpha1.ScaledObjectAuthRef, scaledObject *keda_v1alpha1.ScaledObject, resolveEnv func(string, string) string) (map[string]string, string) {
+	result := make(map[string]string)
+	podIdentity := ""
+
+	if triggerAuthRef.Name != "" {
+		namespace := triggerAuthRef.Namespace
+		if namespace == "" {
+			namespace = scaledObject.Namespace
+		}
+
+		triggerAuth, err := h.kedaClient.KedaV1alpha1().TriggerAuthentications(namespace).Get(triggerAuthRef.Name, meta_v1.GetOptions{})
+		if err != nil {
+			log.Errorf("Error getting triggerAuth %s in namespace %s. Error: %s", triggerAuthRef.Name, namespace, err.Error())
+		} else {
+			podIdentity = string(triggerAuth.Spec.PodIdentity.Provider)
+			if triggerAuth.Spec.Env != nil {
```

Comment:
Do we try environment variables before secret ref? Given the latter are more secure we might want to check those first?

Will probably never happen but if both are configured, I'd expect it to use the most secure one by default, no?

Reply:
Alternative is to throw an exception if more than one is defined?

Reply:
Secrets take precedence over environment in this implementation. And I'd rather not throw errors if the state can be reconciled. See https://github.com/kedacore/keda/issues/379 for example. I think we should make all environment resolution best-effort, then let the scaler fail the connection if it can't make it.

Reply:
Ah you are right, lol I read it too quickly. Thought there was a return in there.

I agree.